test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeGenerator.java

```diff
@@ -419,6 +436,88 @@ private static void generateSetter(ClassWriter cw, String[] nameAndSigValue, Str
 		mv.visitEnd();
 	}
 	
+	private static void generateSetterStatic(ClassWriter cw, String[] nameAndSigValue, String className) {
+		boolean doubleDetected = false;
+		MethodVisitor mv = cw.visitMethod((ACC_PUBLIC | ACC_STATIC), "setStatic" + nameAndSigValue[0], "(Ljava/lang/Class;" + nameAndSigValue[1] + ")V", null, null);
```

Comment:
also, the class  param is not needed since the the putstatic fieldref knows which class it is referring to

Reply:
youll need to make similar changes for the getter

Reply:
I presume by class param, you mean `Ljava/lang/Class;`? When I removed it, I'm getting a NoSuchMethod error regarding the setStatictri method.

Reply:
yes, you need to update the generic setter as well